src/types.jl

```diff
@@ -97,7 +90,19 @@ struct Acb <: Number
     end
 end
 
-for (T, prefix) in ((Mag, :mag), (Arf, :arf), (Arb, :arb), (Acb, :acb))
+
+struct AcbMatrix <: AbstractMatrix{Acb}
+    acb_mat::acb_mat_struct
+    prec::Int
+
+    function AcbMatrix(m::Integer, n::Integer; prec::Integer = DEFAULT_PRECISION[])
+        res = new(acb_mat_struct(m, n), prec)
+        return res
+    end
+end
```

Comment:
We can also add a construct taking an `acb_mat_struct` as is the case for the other types.
```
    function AcbMatrix(A::acb_mat_struct;
                       prec::Integer=DEFAULT_PRECISION[],
                       shallow::Bool = false)
        if shallow
            res = new(A, prec)
        else
            res = AcbMatrix(A.r, A.c, prec = prec)
            set!(res, A)
        end

        return res
    end
```
should work. I can never decide on if I want to call matrices `A` or `M`, Julia seems to favor `A` in general.

Reply:
Sure, but I am not sure how much this is needed. The shallow `Acb` operators are basically only there to support non-allocating indexing of matrices, right?

Reply:
That's true, it might not be very useful. Then we can skip it for now since it's trivial to add later if we realize it's useful.